orderer/common/cluster/clusterservice_test.go

```diff
@@ -0,0 +1,200 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package cluster_test
+
+import (
+	"context"
+	"crypto/tls"
+	"io"
+	"testing"
+
+	"github.com/hyperledger/fabric-protos-go/msp"
+	"github.com/hyperledger/fabric-protos-go/orderer"
+	"github.com/hyperledger/fabric/common/flogging"
+	"github.com/hyperledger/fabric/common/metrics/disabled"
+	"github.com/hyperledger/fabric/orderer/common/cluster"
+	"github.com/hyperledger/fabric/orderer/common/cluster/mocks"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/peer"
+)
+
+var (
+	nodeAuthrequest = &orderer.ClusterNodeServiceStepRequest{
+		Payload: &orderer.ClusterNodeServiceStepRequest_NodeAuthrequest{
+			NodeAuthrequest: &orderer.NodeAuthRequest{
+				Channel: "mychannel",
+			},
+		},
+	}
+	nodeConsensusRequest = &orderer.ClusterNodeServiceStepRequest{
+		Payload: &orderer.ClusterNodeServiceStepRequest_NodeConrequest{
+			NodeConrequest: &orderer.NodeConsensusRequest{
+				Payload: []byte{1, 2, 3},
+			},
+		},
+	}
+	nodeInvalidRequest = &orderer.ClusterNodeServiceStepRequest{
+		Payload: &orderer.ClusterNodeServiceStepRequest_NodeConrequest{
+			NodeConrequest: nil,
+		},
+	}
+)
+
+type AuthVerifier struct {
+	cluster.ClusterService
+}
+
+func (cs *AuthVerifier) VerifyAuthRequest(orderer.ClusterNodeService_StepServer, *orderer.ClusterNodeServiceStepRequest, cluster.TLSSessionBindingGetter, cluster.SignatureVerifier) (string, error) {
+	return "mychannel", nil
+}
+
```

Comment:
The scope of these tests is rather limited.

We should test also by spinning up nodes with real communication in a similar fashion to what is done in `comm_test.go`.

One major problem that I believe that is not addressed in the current production code, and would have surely been caught had the tests been fuller in scope, is that `comm.go` which manages egress communication, still uses TLS pinning for authentication. 

We need code that ensures that upon a reconfiguration, the TLS connections to nodes that are no longer needed are closed, and TLS connections to nodes that are required but are not connected to, would be initiated.